### PR TITLE
Deploy dependent framework packages when registering from layout

### LINKED ISSUE #5179: When deploying an app from layout, we should deploy dependencies
The current deploy from layout codepath assumes that all dependencies (e.g. MUX 2.4, .net / VCLibs) are installed but this isn't true in a clean machine. Installing apps from the store might bring these dependencies onto the box but we should do so if this is a clean machine too.

Otherwise the app fails to deploy when doing `run-windows`.

Fix: when deploying the app from layout we should also deploy something like this first:
`Add-AppxPackage windows\ReactUWPTestApp\AppPackages\ReactUWPTestApp_1.0.0.0_x86_Debug_Test\Dependencies\x86\*`

### PR DESCRIPTION
**Pull request by asklar**
Fixes #5179 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/5454)

### CODE REVIEW

File: vnext/local-cli/src/runWindows/utils/deploy.ts [hunk 1]
```diff
@@ -259,6 +259,18 @@ export async function deployToDesktop(
       verbose,
     );
   } else {
+    // Install the app package's dependencies before attempting to deploy.
+    const dependencies = path.join(
+      appPackageFolder,
+      'Dependencies',
```
Comment (NickGerleman) [inline vnext/local-cli/src/runWindows/utils/deploy.ts:265]: Where does this folder come from?

Comment (asklar) [reply vnext/local-cli/src/runWindows/utils/deploy.ts:265]: The UWP project system creates it.

File: vnext/local-cli/src/runWindows/utils/commandWithProgress.ts [hunk 2]
```diff
@@ -49,7 +49,7 @@ export async function runPowerShellScriptFunction(
 ) {
   try {
     const printException = verbose ? '$_;' : '';
-    const importScript = script ? `Import-Module "${script}"; ` : '';
+    const importScript = script !== '' ? `Import-Module "${script}"; ` : '';
```
Comment (NickGerleman) [inline vnext/local-cli/src/runWindows/utils/commandWithProgress.ts:52]: If you want to allow null, you can add it as an allowed type

Comment (NickGerleman) [reply vnext/local-cli/src/runWindows/utils/commandWithProgress.ts:52]: I.e. "script: string | null"

**Pull request merged**